Content.Goobstation.Shared/MartialArts/SharedMartialArtsSystem.PlasmaFist.cs

```diff
@@ -0,0 +1,178 @@
+            {
+                entities.Add(uid);
+            }
+        }
+
+        foreach (var entity in entities)
+        {
+            if (!physicsQuery.TryGetComponent(entity, out var physics))
+                continue;
+
+            if (movedByPressureQuery.TryGetComponent(entity, out var movedPressure) && !movedPressure.Enabled) //Ignore magboots users
+                continue;
+
+            // Calculate the direction from the source to the entity.
+            var entityXform = _entityManager.GetComponent<TransformComponent>(entity);
+            var direction = _transform.GetMapCoordinates(entity).Position - position;
+            var distance = direction.Length();
+
+            // Normalize the direction.
+            if (distance > 0)
+            {
+                direction /= distance;
+            }
+            else
+            {
+                // If the entity is at the exact same position, apply a random direction.
+                direction = _random.NextVector2().Normalized();
+            }
+            // Apply the impulse.
+            _entityManager.System<SharedPhysicsSystem>().ApplyLinearImpulse(entity, direction * 1000, body: physics);
+        }
+        var ev = new MartialArtSaying("plasmafist-saying-tornado");
+        RaiseLocalEvent(ent, ev);
+        ComboPopup(ent, target, proto.Name);
+    }
+
+    private void OnThrowback(Entity<CanPerformComboComponent> ent, ref ThrowbackPerformedEvent args)
+    {
+        if (!_proto.TryIndex(ent.Comp.BeingPerformed, out var proto)
+            || !TryUseMartialArt(ent, proto, out var target, out var downed))
+            return;
+
+        var mapPos = _transform.GetMapCoordinates(ent).Position;
+        var hitPos = _transform.GetMapCoordinates(target).Position;
+        var dir = hitPos - mapPos;
+        dir *= 1f / dir.Length();
+
+        if (TryComp<PullableComponent>(target, out var pullable))
+            _pulling.TryStopPull(target, pullable, ent, true);
+        _grabThrowing.Throw(target, ent, dir, 50f);
+
+        ComboPopup(ent, target, proto.Name);
+
+        var items = _hands.EnumerateHeld(ent);
+        foreach (var item in items)
+        {
+            if (!_hands.TryDrop(ent, item))
+                break;
+            _throwing.TryThrow(item, dir, 50f, ent, 0);
+        }
+        items = _hands.EnumerateHeld(target);
+        foreach (var item in items)
+        {
+            if (!_hands.TryDrop(target, item))
+                break;
+            _throwing.TryThrow(item, dir, 50f, ent, 0);
+        }
+    }
+    private void OnThePlasmaFist(Entity<CanPerformComboComponent> ent, ref ThePlasmaFistPerformedEvent args)
+    {
+        if (!_proto.TryIndex(ent.Comp.BeingPerformed, out var proto)
+            || !TryUseMartialArt(ent, proto, out var target, out var downed))
+            return;
+        if (TryComp<BodyComponent>(ent, out BodyComponent? outBody) && outBody is not null)
```

Comment:
```suggestion
        if (TryComp<BodyComponent>(ent, out BodyComponent? outBody))
```
TryComp ensures that returned component is not null if method returns true